src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DrawioWebView.kt

```diff
@@ -34,8 +36,21 @@ class DrawioWebView(lifetime: Lifetime) : BaseDrawioWebView(lifetime) {
         }
     }
 
+    override fun handleResponse(response: IncomingMessage.Response) {
+        when(response) {
+            is IncomingMessage.Response.Export -> {
+                val payload = response.data.split(",")[1]
+                val decodedBytes = Base64.getDecoder().decode(payload)
+                val svg = String(decodedBytes)
+                _xmlContent.set(svg)
+            }
+        }
+    }
     fun loadXmlLike(xmlLike: String) {
         _xmlContent.set(null) // xmlLike is not xml
         send(OutgoingMessage.Event.Load(xmlLike, 1))
     }
+    fun exportSvg() {
+        send(OutgoingMessage.Request.Export(OutgoingMessage.Request.Export.XMLSVG))
+    }
```

Comment:
This send method returns a promise! You can use `.then { response -> (response as IncomingMessage.Response.Export).data }` and return a mapped promise.

Reply:
this should also be fixed